File: python/src/winload/emoji.py
"""Emoji decoration helpers for CLI-facing labels."""

from __future__ import annotations


_ICONS = {
    "description": "🌐",
    "help_interval": "⏱️",
    "help_average": "📊",
    "help_device": "🖧",
    "help_title": "🏷️",
    "help_title_align": "↔️",
    "help_debug_info": "🔍",
    "help_emoji": "✨",
    "help_unicode": "█",
    "help_unit": "🧮",
    "help_bar_style": "🎛️",
    "help_in_color": "⬇️",
    "help_out_color": "⬆️",
    "help_max_mode": "📈",
    "help_max_half_life": "⏳",
    "help_max_y_value": "📏",
    "help_no_graph": "🚫",
    "help_hide_separator": "➖",
    "help_version": "🏷️",
    "help_no_color": "⚫",
    "help_npcap": "🪟",
    "help_netlink": "🔗",
    "help_lang": "🌐",
    "help_system_info": "🖥️",
}


def decorate(enabled: bool, key: str, text: str) -> str:
    """Prefix text with the icon for key when emoji output is enabled."""
    if not enabled:
        return text
    icon = _ICONS.get(key)
    if not icon:
        return text
    return f"{icon} {text}"

File: rust/src/emoji.rs
// ✨ Decorates CLI-facing labels with optional emoji icons.
pub fn decorate(enabled: bool, key: &str, text: impl Into<String>) -> String {
    let text = text.into();
    if !enabled {
        return text;
    }

    icon(key)
        .map(|icon| format!("{icon} {text}"))
        .unwrap_or(text)
}

pub fn icon(key: &str) -> Option<&'static str> {
    match key {
        "description" => Some("🌐"),
        "help_interval" => Some("⏱️"),
        "help_average" => Some("📊"),
        "help_device" => Some("🖧"),
        "help_title" => Some("🏷️"),
        "help_title_align" => Some("↔️"),
        "help_debug_info" => Some("🔍"),
        "help_emoji" => Some("✨"),
        "help_unicode" => Some("█"),
        "help_unit" => Some("🧮"),
        "help_bar_style" => Some("🎛️"),
        "help_in_color" => Some("⬇️"),
        "help_out_color" => Some("⬆️"),
        "help_max_mode" => Some("📈"),
        "help_max_half_life" => Some("⏳"),
        "help_max_y_value" => Some("📏"),
        "help_no_graph" => Some("🚫"),
        "help_hide_separator" => Some("➖"),
        "help_no_color" => Some("⚫"),
        "help_npcap" => Some("🪟"),
        "help_netlink" => Some("🔗"),
        "help_lang" => Some("🌐"),
        "help_shortcuts_title" => Some("⌨️"),
        "shortcut_previous_device" => Some("⬅️"),
        "shortcut_next_device" => Some("➡️"),
        "shortcut_toggle_debug" => Some("🔧"),
        "shortcut_toggle_separator" => Some("➖"),
        "shortcut_toggle_color" => Some("🎨"),
        "shortcut_cycle_graph_style" => Some("📈"),
        "shortcut_toggle_x_axis" => Some("⏱️"),
        "shortcut_cycle_y_axis" => Some("📏"),
        "shortcut_quit" => Some("🚪"),
        "help_system_info" => Some("🖥️"),
        _ => None,
    }
}
